plugins/tableselection/plugin.js

```diff
@@ -443,37 +459,159 @@
+	// @returns {HTMLElement[]}
+	TableSelection.prototype.getTableMap = function() {
+		function getRealCellPosition( cell ) {
+			var table = cell.getAscendant( 'table' ),
+				rowIndex = cell.getParent().$.rowIndex,
+				map = CKEDITOR.tools.buildTableMap( table ),
+				i;
+
+			for ( i = 0; i < map[ rowIndex ].length; i++ ) {
+				if ( new CKEDITOR.dom.element( map[ rowIndex ][ i ] ).equals( cell ) ) {
+					return i;
+				}
+			}
+		}
 
+		var startIndex = getCellColIndex( this.cells.first, true ),
+			endIndex = getRealCellPosition( this.cells.last );
+
+		return CKEDITOR.tools.buildTableMap(  this._getTable(), this.rows.first.$.rowIndex, startIndex,
+			this.rows.last.$.rowIndex, endIndex );
+	};
+
+	TableSelection.prototype._getTable = function() {
+		return this.rows.first.getAscendant( 'table' );
+	};
+
+	// @param {Number} count Number of rows to be inserted.
+	// @param {Boolean} [insertBefore=false] If set to `true` new rows will be prepended.
+	// @param {Boolean} [clearSelection=false] If set to `true`, it will set selected cells to the one inserted.
+	TableSelection.prototype.insertRow = function( count, insertBefore, clearSelection ) {
+		if ( typeof count === 'undefined' ) {
+			count = 1;
+		} else if ( count <= 0 ) {
+			return;
+		}
+
+		var cellIndexFirst = this.cells.first.$.cellIndex,
+			cellIndexLast = this.cells.last.$.cellIndex,
+			selectedCells = clearSelection ? [] : this.cells.all,
+			row,
+			newCells;
+
+		for ( var i = 0; i < count; i++ ) {
+			// In case of clearSelection we need explicitly use cached cells, as selectedCells is empty.
+			row = insertRow( clearSelection ? this.cells.all : selectedCells, insertBefore );
+
+			// Append cells from added row.
+			newCells = CKEDITOR.tools.array.filter( this._nodeListToArray( row.find( 'td, th' ) ), function( cell ) {
+				return clearSelection ?
+					true : cell.$.cellIndex >= cellIndexFirst && cell.$.cellIndex <= cellIndexLast;
+			} );
+
+			// setSelectedCells will take care of refreshing the whole state at once.
+			if ( insertBefore ) {
+				selectedCells = newCells.concat( selectedCells );
+			} else {
+				selectedCells = selectedCells.concat( newCells );
+			}
+		}
+
+		this.setSelectedCells( selectedCells );
+	};
+
+	// @param {Number} count Number of columns to be inserted.
+	TableSelection.prototype.insertColumn = function( count ) {
+		if ( typeof count === 'undefined' ) {
+			count = 1;
+		} else if ( count <= 0 ) {
+			return;
+		}
+
+		var selectedCells = this.cells.all;
+
+		for ( var i = 0; i < count; i++ ) {
+			// Prepend added cells, then pass it to setSelectionCells so that it will
+			// take care of refreshing the whole state.
+			selectedCells = selectedCells.concat( insertColumn( selectedCells ) );
+		}
+
+		this.setSelectedCells( selectedCells );
+	};
+
+	// Sorts given arr according to DOM position.
+	//
+	// @param {CKEDITOR.dom.node[]} arr
+	TableSelection.prototype._arraySortByDOMOrder = function( arr ) {
```

Comment:
It looks like I have lost it's usage during cleanups. It was introduced specifically for `insertColumn` call, as it [just appends created `td` elements into `selectedCells`](https://github.com/ckeditor/ckeditor-dev/blob/501b8b2/plugins/tableselection/plugin.js#L586), so this will append `n` columns from each row.

However I think it makes a good idea to put it in `_reset`. It's not a super performance-intensive function so that's going to be fine.